### DISCUSSION TITLE
[OP Stack Deployment] Questions about being a part of superchain

### DISCUSSION THREAD

**Discussion by shyam-patel-kira** [ethereum-optimism/developers#497] [Deploying the OP Stack]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Forgive me if this is not a right place to ask this, The questions assume that I’m building a custom L2 rollup using Op-stack and the chain wants to be a part of the Superchain.

> The chain’s genesis state and predeploy (0x420000...) contracts are standard, uniform implementations with no malicious code.

1. Can we add one or more pre-deploy contracts, which is not malicious but there to have some additional functionality for the new rollup? If one does add one or more predeploys does it still qualify as a standard rollup?

> A standard fee split to the Collective is enforced by smart contracts, as specified [somewhere] and implemented by [some contract].

2. Even though currently all chains use ETH as their gas token, if someone wants to use a custom native gas token in future, or let’s the new rollup has a feature of custom gas token and then would this affect the stance of the new rollup or existing rollup as a standard Rollup chain?

3. Adding to the above question how would fee split work in that case if custom gas token is implemented/used? Emphasis on custom gas token because let’s say a chain is made for some specific usecase other than the primary use-cases that current chains have, something like redstone for gaming or some chain for other real-world use-case. How would this affect it(the stance of being a part of superchain)?

Thanks in advance!

**Comment by hash1go** [comment]
The conditions to meet the standard rollup requirements are currently being formulated and, I believe, have not been published yet.
Also, the details regarding profit sharing when using custom gas tokens have not been decided yet I think.

Hello @sbvegan 
Is there any public information available regarding this?

**Reply by sbvegan** [reply]
Yeah this is a question about the standard chain charter, you can see it [here](https://gov.optimism.io/t/season-6-draft-standard-rollup-charter/8135/7)

**Comment by sbvegan** [comment]
The baseline requirement to join the Superchain is to have an agreement in place with the Optimism Foundation and be contributing some of your sequencer revenue back to the Optimism Collective. Any OP Stack chain is eligible to join, standard or not, if they meet that criteria. The Standard Chain Charter, is for standard chains in the superchain.
